File: minnow2/pong.cpp
#include <phish.hpp>

void message_callback(uint32_t parts)
{
  phish::pack(phish::unpack<std::string>());
  phish::send();
}

int main(int argc, char* argv[])
{
  phish::init(argc, argv);
  phish::input(0, message_callback, phish::close);
  phish::output(0);
  phish::check();
  phish::loop();

  return 0;
}

File: example2/pong.py
import phish

def message_callback(message):
  phish.send(message)

def closed_callback():
  phish.close()

phish.init()
phish.enable_input_port(0, message_callback, closed_callback)
phish.enable_output_port(0)
phish.check()
phish.loop()

